ajax-reddit-slideshow/script.js

```diff
@@ -0,0 +1,104 @@
+$(function() { 
+  // attach the form submission to the search function
+  $("#search-form").on('submit', search);
+});
+
+function search(event) {
+  // Stop the form from changing the page.
+  event.preventDefault();
+
+    alert("You just did a search!");
+
+
+  clearSearchResults();
+
+
+  // Get the users search input and save it in a variable.
+  // Use the input placeholder value (like "kittens") as a default value.
+  var userQuery = undefined;
+
+  console.log("searching for:", userQuery);
+
+  $.get('http://www.reddit.com/search.json', {
+    q: $("#query").val()
+  }).done(function(response) {
+    console.log(response);
+
+    var results = response.data.children
+
+    for (var i = 0; i < results.length; i++) {
+      // console.log(results[i]);
+      var result = results[i].data
+      addSearchResult(result)
+
+    $( "#prev_image" ).click(function(){
+       prev();
+    });
+    $( "#next_image" ).click(function(){
+       next();
+    });
+
+    }
+
+
+  });
+
+}
+
+
+
+// // Clear previous search results.
+function clearSearchResults() {
+  $("#images").html("");
+}
+
+function addSearchResult(result) {
+
```

Comment:
One option here is to push the urls into a global array and create a counter that we can then use to display images. Either through a set interval or a setTimeout implementation.